Return unavailable if Govee BLE sensor is in error

PierreAronnax opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Allow Govee BLE sensors to report as unavailable when the sensor itself reports an error.
Needs govee-ble dependency bump.

changelog: https://github.com/Bluetooth-Devices/govee-ble/compare/v0.19.3...v0.21.0

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/Bluetooth-Devices/govee-ble/issues/51 https://github.com/Bluetooth-Devices/govee-ble/pull/55
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

PierreAronnax committed:
Return available False if Govee BLE sensor is in error

PierreAronnax committed:
add h5178 error test

bdraco commented on homeassistant/components/govee_ble/sensor.py:
```diff
@@ -122,6 +123,13 @@ class GoveeBluetoothSensorEntity(
+    @property
+    def available(self) -> bool:
+        """Return False if sensor is in error."""
+        if self.processor.entity_data.get(self.entity_key) == ERROR:
+            return False
+        return super().available
```
```suggestion
        return self.processor.entity_data.get(self.entity_key) != ERROR and super().available
```

bdraco reviewed (changes requested):
Thank you for your contribution thus far! 🎖 Since this is a significant contribution, we would appreciate you'd added yourself to the list of code owners for this integration. ❤️ 

Please, add your GitHub username to the `manifest.json` of this integration.

For more information about "code owners", see: [Architecture Decision Record 0008: Code owners](https://github.com/home-assistant/architecture/blob/master/adr/0008-code-owners.md).

PierreAronnax committed:
Return available as oneliner

PierreAronnax committed:
Add @PierreAronnax to govee_ble codeowners

PierreAronnax committed:
update h5075 test

PierreAronnax committed:
add str to types

homeassistant/components/govee_ble/sensor.py:130: error: Non-overlapping equality check (left operand type: "Union[float, int, None]", right operand type: "str")  [comparison-overlap]
homeassistant/components/govee_ble/sensor.py:137: error: Incompatible return value type (got "Union[float, int, str, None]", expected "Union[int, float, None]")  [return-value]

PierreAronnax commented on homeassistant/components/govee_ble/sensor.py:
```diff
@@ -116,13 +117,21 @@ async def async_setup_entry(
 class GoveeBluetoothSensorEntity(
     PassiveBluetoothProcessorEntity[
-        PassiveBluetoothDataProcessor[Optional[Union[float, int]]]
+        PassiveBluetoothDataProcessor[Optional[Union[float, int, str]]]
```
Without this we get
`homeassistant/components/govee_ble/sensor.py:130: error: Non-overlapping equality check (left operand type: "Union[float, int, None]", right operand type: "str")  [comparison-overlap]`

Or do you prefer
`self.processor.entity_data.get(self.entity_key) != ERROR  # type: ignore[comparison-overlap]` ?

bdraco replied:
better to do it as you have it and avoid the ignore

bdraco committed:
bump govee-ble to 0.21.0

bdraco committed:
use const

bdraco committed:
Merge branch 'dev' into govee_ble_error

bdraco commented:
Thanks @PierreAronnax

bdraco commented:
Tested 👍

PierreAronnax commented:
Thanks!
